server/build/webpack/utils.js

```diff
@@ -50,12 +52,20 @@ export function createEntry (filePath, {name, pageExtensions} = {}) {
 }
 
 // Convert page paths into entries
-export function getPageEntries (pagePaths, {isServer = false, pageExtensions} = {}) {
+export function getPageEntries (pagePaths, opts) {
+  return getMultiRootPageEntries([{ root: '.', pages: pagePaths }], opts)
+}
+
+function getMultiRootPageEntries (pagePathsByRoot, {isServer = false, pageExtensions} = {}) {
   const entries = {}
 
-  for (const filePath of pagePaths) {
-    const entry = createEntry(filePath, {pageExtensions})
-    entries[entry.name] = entry.files
+  for (const { root, pages } of pagePathsByRoot) {
+    for (const filePath of pages) {
+      const entry = createEntry(filePath, {pageExtensions, root})
+      if (!(entry.name in entries)) {
+        entries[entry.name] = entry.files
+      }
```

Comment:
Looks like here it would loop through different rootPaths and first found entry will be used as the result,
although script will continue to loop through the rest of the list. Correct?

And it seems to be a bit different from behavior when exporting, that it will keep looping and overwriting found files.

So for the pages – first rootPaths will take precedence, but for static files the last rootPaths will override files that came before.

I think it'd be better to make it consistent and to mimic object merge behavior, so that paths on the right will override paths to the left of them.

Reply:
In both cases the first root paths take precedence. In the static file export code, we loop backwards to make sure that the files copied over are overwritten by files in the directory at the start of the array.